src/test/blockencodings_tests.cpp

```diff
@@ -386,7 +387,9 @@ BOOST_AUTO_TEST_CASE(TransactionsRequestDeserializationOverflowTest) {
         BOOST_CHECK(0);
     } catch(std::ios_base::failure &) {
         // deserialize should fail
+        exception_thrown = true;
     }
+    BOOST_CHECK(exception_thrown);
```

Comment:
Could you please explain the control flow where this check fails, but the `BOOST_CHECK(0);` above passes?

Reply:
There shouldn't exist such a case: they are meant to be equivalent.

IIRC the re-formulation was needed to avoid having the test runner warn that the test case succeeded without executing any succeeding checks (or something along those lines).

The re-formulated version made sure that a succeeding check (`BOOST_CHECK(exception_thrown); `) was executed as part of the test case.

However, when re-running without this I don't get this warning so it must have been under some exotic command-line option. I'm unable to reproduce the condition this was meant to solve.

I've now removed this since it isn't needed to silence `src/test/test_bitcoin` which is the goal of this PR :-)

Reply:
After some additional testing I was able to reproduce the warnings that these formulations were meant to silence:

```
Test case blockencodings_tests/TransactionsRequestDeserializationOverflowTest did not check any assertions
Test case util_tests/util_criticalsection did not check any assertions
```

Re-adding those re-formulations to get rid of these warnings.

Reply:
Why would you need to keep a variable for this?

Wouldn't `BOOST_CHECK(true); // Explain why this is needed` be sufficient?

Reply:
Yes, absolutely. Reformulated to an equivalent version not introducing any new variables :-)